src/qwed_open_responses/guards/tool_guard.py

```
@@ -134,6 +134,16 @@ def check(
         # Check each tool call
         for call in tool_calls:
             tool_name = call.get("tool_name") or call.get("name")
+
+            # Unrecognized envelope (#28): fail closed, never pass silently.
+            if tool_name is None and call.get("type") == "__unrecognized__":
+                return self.fail_result(
+                    "BLOCKED: Response contains tool-like content in an unrecognized format. "
+                    f"Supported shapes: type=tool_call, tool_calls[], choices[].message.tool_calls[], "
```

Ruff (0.16.2)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)